### DISCUSSION TITLE
PHP 8 support in laravel-analytics 4

### DISCUSSION THREAD

**Discussion by azizullahsaeidi** [freekmurze/laravel-analytics#409] [Q&A]
I am using laravel 8 & and php 8 but I am getting the bellow error during installation. 
 
![Screenshot from 2021-06-14 22-18-55](https://user-images.githubusercontent.com/18898699/121936932-63824100-cd5f-11eb-84c0-a303fabb1d48.png)

**Comment by jeffochoa** [comment]
I was having the same issue, checking my `composer.lock` and I found that the `guzzlehttp\psr7` package was locked to version `2.0.0`, and when you try to install the `spatie/laravel-analytics` package it fails because the `googleapis/google-api-php-client` requires `"guzzlehttp/psr7": "^1.2"`, see https://github.com/googleapis/google-api-php-client/blob/master/composer.json#L16

I'm using Laravel 8.4 which uses `guzzlehttp/guzzle` 7.3.0`, which should work fine with `guzzlehttp/psr7` ^1.7 as well. From my composer.lock:

```
        {
            "name": "guzzlehttp/guzzle",
            "version": "7.3.0",
            ...
            "require": {
               ....
                "guzzlehttp/psr7": "^1.7 || ^2.0"
```

That means that even though the default Laravel install is locking `guzzlehttp/psr7` to 2.0.0, it should be safe to downgrade to `^1.7`.

## Solution for Laravel 8.4

1. Downgrade `guzzlehttpsr7`

```bash
composer require guzzlehttp/psr7 ^1.7
```

2. Install Laravel Analytics

```bash
composer require spatie/laravel-analytics
```

The long-term solution would be to update the dependencies on the `googleapis/google-api-php-client` package. See https://github.com/googleapis/google-api-php-client/blob/master/composer.json


I hope that helps!

Jeff.
